### ISSUE TITLE
Questions about synthetic dataset

### ISSUE THREAD

**Issue by Jerry-hyl** [gydpku/Data_Synthesis_RL#5]
When generating the RL training dataset as described in the paper, I didn’t see any explanation of how the answers (ground truth) were generated. The focus was mainly on the automated scaling of instructions and inputs. So, how is the ground truth answer generated?

**Comment by hexixiang** [comment]
Because the reward function in reinforcement learning only needs to compare whether the math problem solution is correct, there is no need to synthesize or rewrite the answer.

**Comment by gydpku** [comment]
Hi Jerry, we generate both the input and answer by using GPT4o.

**Comment by nnWhisperer** [comment]
> Hi Jerry, we generate both the input and answer by using GPT4o.

IMHO, considering that both the input and answers are generated by using GPT4o, this work shouldn't be considered doing RL but this is distillation from a better model to a smaller one, iteratively in a loop.
May be what is missing is some kind of an environment where the validity is checked and reward is generated accordingly.
